File: android/MeshTRX/app/src/main/java/com/meshtrx/app/ui/Colors.kt
package com.meshtrx.app.ui

import android.graphics.Color

object Colors {
    val bgPrimary     = Color.parseColor("#141414")
    val bgSurface     = Color.parseColor("#1a1a1a")
    val bgElevated    = Color.parseColor("#222222")
    val bgBorder      = Color.parseColor("#2a2a2a")
    val borderSubtle  = Color.parseColor("#333333")
    val borderMuted   = Color.parseColor("#252525")

    val greenAccent   = Color.parseColor("#4ade80")
    val greenBg       = Color.parseColor("#1a3a1a")
    val greenBorder   = Color.parseColor("#2a5a2a")
    val greenDim      = Color.parseColor("#2d5a2d")

    val blueAccent    = Color.parseColor("#5ba3e8")
    val blueBg        = Color.parseColor("#1e3a5f")
    val blueBorder    = Color.parseColor("#2a5a8f")

    val redAccent     = Color.parseColor("#f87171")
    val redTx         = Color.parseColor("#ff6b6b")
    val redBg         = Color.parseColor("#3a0e0e")
    val redBorder     = Color.parseColor("#6b1a1a")
    val redTxBg       = Color.parseColor("#4a0a0a")
    val redTxBorder   = Color.parseColor("#aa2a2a")

    val amberAccent   = Color.parseColor("#f59e0b")
    val amberBg       = Color.parseColor("#2a1f0a")

    val textPrimary   = Color.parseColor("#e8e8e8")
    val textSecondary = Color.parseColor("#cccccc")
    val textMuted     = Color.parseColor("#888888")
    val textDim       = Color.parseColor("#555555")
    val textLabel     = Color.parseColor("#444444")

    val navBg         = Color.parseColor("#111111")
    val navBorder     = Color.parseColor("#222222")

    fun rssiColor(rssi: Int): Int = when {
        rssi >= -60 -> greenAccent
        rssi >= -90 -> amberAccent
        else -> redAccent
    }

    fun rssiBg(rssi: Int): Int = when {
        rssi >= -60 -> greenBg
        rssi >= -90 -> amberBg
        else -> redBg
    }
}

File: ios/MeshTRX/MeshTRX/Utils/Colors.swift
import SwiftUI

enum AppColors {
    // Background
    static let bgPrimary    = Color(hex: 0x141414)
    static let bgSurface    = Color(hex: 0x1a1a1a)
    static let bgElevated   = Color(hex: 0x222222)
    static let bgBorder     = Color(hex: 0x2a2a2a)
    static let borderSubtle = Color(hex: 0x333333)
    static let borderMuted  = Color(hex: 0x252525)

    // Green
    static let greenAccent  = Color(hex: 0x4ade80)
    static let greenBg      = Color(hex: 0x1a3a1a)
    static let greenBorder  = Color(hex: 0x2a5a2a)
    static let greenDim     = Color(hex: 0x2d5a2d)

    // Blue
    static let blueAccent   = Color(hex: 0x5ba3e8)
    static let blueBg       = Color(hex: 0x1e3a5f)
    static let blueBorder   = Color(hex: 0x2a5a8f)

    // Red
    static let redAccent    = Color(hex: 0xf87171)
    static let redTx        = Color(hex: 0xff6b6b)
    static let redBg        = Color(hex: 0x3a0e0e)
    static let redBorder    = Color(hex: 0x6b1a1a)
    static let redTxBg      = Color(hex: 0x4a0a0a)
    static let redTxBorder  = Color(hex: 0xaa2a2a)

    // Amber
    static let amberAccent  = Color(hex: 0xf59e0b)
    static let amberBg      = Color(hex: 0x2a1f0a)

    // Text
    static let textPrimary   = Color(hex: 0xe8e8e8)
    static let textSecondary = Color(hex: 0xcccccc)
    static let textMuted     = Color(hex: 0x888888)
    static let textDim       = Color(hex: 0x555555)
    static let textLabel     = Color(hex: 0x444444)

    // Nav
    static let navBg         = Color(hex: 0x111111)
    static let navBorder     = Color(hex: 0x222222)

    static func rssiColor(_ rssi: Int) -> Color {
        if rssi >= -60 { return greenAccent }
        if rssi >= -90 { return amberAccent }
        return redAccent
    }

    static func rssiBg(_ rssi: Int) -> Color {
        if rssi >= -60 { return greenBg }
        if rssi >= -90 { return amberBg }
        return redBg
    }
}

// MARK: - Color hex init

extension Color {
    init(hex: UInt32, alpha: Double = 1.0) {
        self.init(
            .sRGB,
            red: Double((hex >> 16) & 0xFF) / 255,
            green: Double((hex >> 8) & 0xFF) / 255,
            blue: Double(hex & 0xFF) / 255,
            opacity: alpha
        )
    }
}
